File: public/generate_icons.js
const sharp = require('sharp');
const fs = require('fs');
const path = require('path');

const SVG_PATH = path.join(__dirname, 'logo.svg');
const svgBuffer = fs.readFileSync(SVG_PATH);

async function generate() {
  const sizes = [
    { name: 'favicon-16x16.png',        size: 16  },
    { name: 'favicon-32x32.png',        size: 32  },
    { name: 'apple-touch-icon.png',     size: 180 },
    { name: 'android-chrome-192x192.png', size: 192 },
    { name: 'android-chrome-512x512.png', size: 512 },
  ];

  for (const { name, size } of sizes) {
    await sharp(svgBuffer)
      .resize(size, size, { fit: 'contain', background: { r: 0, g: 0, b: 0, alpha: 0 } })
      .png()
      .toFile(path.join(__dirname, name));
    console.log(`  ✓ ${name} (${size}x${size})`);
  }

  // favicon.ico — embed 32x32 PNG (browsers accept PNG-in-ICO)
  await sharp(svgBuffer)
    .resize(32, 32, { fit: 'contain', background: { r: 0, g: 0, b: 0, alpha: 0 } })
    .png()
    .toFile(path.join(__dirname, 'favicon.ico'));
  console.log('  ✓ favicon.ico (32x32)');

  // Social image: 1200x630 warm cream background, logo centered
  const logoSize = 420;
  const logoBuffer = await sharp(svgBuffer)
    .resize(logoSize, logoSize, { fit: 'contain', background: { r: 0, g: 0, b: 0, alpha: 0 } })
    .png()
    .toBuffer();

  const x = Math.round((1200 - logoSize) / 2);
  const y = Math.round((630 - logoSize) / 2);

  await sharp({
    create: {
      width: 1200,
      height: 630,
      channels: 4,
      background: { r: 255, g: 248, b: 245, alpha: 1 },
    },
  })
    .composite([{ input: logoBuffer, left: x, top: y }])
    .png()
    .toFile(path.join(__dirname, 'social-networks-1200x630.png'));
  console.log('  ✓ social-networks-1200x630.png');

  console.log('\nAll icons generated!');
}

generate().catch(console.error);

File: public/generate_icons.py
#!/usr/bin/env python3
"""
Generate all favicon/icon sizes from logo.svg
Requires: pip install Pillow cairosvg
"""
import subprocess, sys, os

# Install cairosvg if needed
try:
    import cairosvg
except ImportError:
    subprocess.check_call([sys.executable, "-m", "pip", "install", "cairosvg", "--break-system-packages", "-q"])
    import cairosvg

from PIL import Image
import io

SVG_PATH = os.path.join(os.path.dirname(__file__), "logo.svg")
OUT_DIR  = os.path.dirname(__file__)

def svg_to_png(size):
    png_bytes = cairosvg.svg2png(url=SVG_PATH, output_width=size, output_height=size)
    return Image.open(io.BytesIO(png_bytes)).convert("RGBA")

sizes = {
    "favicon-16x16.png":        16,
    "favicon-32x32.png":        32,
    "apple-touch-icon.png":     180,
    "android-chrome-192x192.png": 192,
    "android-chrome-512x512.png": 512,
}

for filename, size in sizes.items():
    img = svg_to_png(size)
    out = os.path.join(OUT_DIR, filename)
    img.save(out, "PNG")
    print(f"  ✓ {filename} ({size}x{size})")

# Generate favicon.ico (multi-size: 16, 32, 48)
ico_images = [svg_to_png(s) for s in [16, 32, 48]]
ico_path = os.path.join(OUT_DIR, "favicon.ico")
ico_images[0].save(ico_path, format="ICO", sizes=[(16,16),(32,32),(48,48)],
                   append_images=ico_images[1:])
print(f"  ✓ favicon.ico (16/32/48)")

# Social image: 1200x630 with logo centered on white/gradient bg
social = Image.new("RGBA", (1200, 630), (255, 248, 245, 255))
logo_big = svg_to_png(380)
x = (1200 - 380) // 2
y = (630 - 380) // 2
social.paste(logo_big, (x, y), logo_big)
social.save(os.path.join(OUT_DIR, "social-networks-1200x630.png"), "PNG")
print(f"  ✓ social-networks-1200x630.png")

print("\nAll icons generated successfully!")
